### ISSUE TITLE
vscode.workspace.applyEdit doesnt auto save changes while refactoring

### ISSUE THREAD

**Issue by ctf0** [microsoft/vscode#168825]
Does this issue occur when all extensions are disabled?: Yes

```
Version: 1.75.0-insider
Commit: f076d553210e9ea65e7bccbfa2f7a786c10fcab2
Date: 2022-12-09T09:18:09.258Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: Yes
```

## Issue
am trying to update php files references on file rename/move using `onDidRenameFiles` + `vscode.workspace.applyEdit` but after the changes are made, the file are keept opened and changes are not saved.

i also have `"files.refactoring.autoSave": true` but its not working.

> Api example: 
> 
> ```js
> let edit = new vscode.WorkspaceEdit
> edit.replace(file.uri, file.range, 'something new')
> 
> await vscode.workspace.applyEdit(edit, { isRefactoring: true })
> ```
> 

## Steps to Reproduce:

1. register a `onDidRenameFiles` listener
2. create a main file to test with and add some data that `vscode.executeReferenceProvider` can pick up
3. create a couple other files with a reference to the main file
4. try to apply the changes on file rename to all the references
5. all files will be updated but will remian opened

**Comment by jrieken** [comment]
We don't save the file that's open in the current editor because the "normal" auto-save will handle that. This was some change but for the short time in which we did save we got negative feedback about this. 

@ctf0 Is that what you are seeing? The active editor isn't being saved but cross file edits work properly?

**Comment by ctf0** [comment]
no, i dont have any opened editors, even the one am renaiming, am doing it through the `explorer context menu > rename`

all the files that being changed are left unsaved & opened
